src/main/cpp/src/SparkResourceAdaptorJni.cpp

```diff
@@ -118,35 +132,40 @@ class spark_resource_adaptor final : public rmm::mr::device_memory_resource {
   std::mutex state_mutex;
   std::map<long, thread_state> threads;
   std::map<long, std::set<long>> task_to_threads;
+  jclass retry_oom_class;
+  jclass split_and_retry_oom_class;
+  jclass cudf_exception_class;
   JavaVM *jvm;
 
+  void throw_java_exception(jclass ex_class, const char* msg) {
+    JNIEnv *env = cudf::jni::get_jni_env(jvm);
+    if (ex_class != nullptr) {
+      env->ThrowNew(ex_class, msg);
```

Comment:
Was the intent of this method to return back to the caller if there was an exception class?  The way it's used seems to imply it's not expecting this method to return but always throw a C++-level exception.

Reply:
No that's not the intent. We want this to throw to java, we also want the C++ code to see an exception, I just got confused with `ThrowNew` thinking it did both. Also realizing that's not what we do in RmmJni.cpp for example, comparing that with what I did here to see where I went wrong.

Reply:
ok should befixed